internal/slice/slice.go

```diff
@@ -3,6 +3,48 @@
 
 package slice
 
+import (
+	"reflect"
+
+	"github.com/google/go-cmp/cmp"
+)
+
 func RemoveFromSlice[A any](slice []A, i int) []A {
 	return append(slice[:i], slice[i+1:]...)
 }
+
+// DeepEqual compares two given slices, a and b, of an arbitrary type and checks the order of elements.
+// It returns true (indicating that the slices are equal) if all of the following conditions are true:
+// - they are both nil or both not nil
+// - they have the same length
+// - they have the same type
+// - all elements of slice a are present in slice b
+// Otherwise, it returns false.
+func DeepEqual[A any](a, b []A) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	if a == nil || b == nil {
+		return true
+	}
+
+	if reflect.TypeOf(a) != reflect.TypeOf(b) {
+		return false
+	}
+
+	for _, va := range a {
+		found := false
+		for _, vb := range b {
+			if cmp.Equal(va, vb) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
```

Comment:
So the logic here means that two slices are equal if they are the same length and contain the same elements in any order, is that right? 

If so, there isn't actually a test case for this, i.e: `{1, 2, 3}` is equal to `{3, 2, 1}`

Reply:
Also, with this logic it doesn't take into account duplicates so this function will consider `{1, 1, 1, 2, 3}` and `{1, 1, 2, 2, 3}` equal. Is that intentional?

Reply:
Last one: Because the loop here is only checking if B contains elements from A, and doesn't consider ordering or duplicates, then we get the following result:

```
{1, 1} equals {1, 5} = true
{1, 5} equals {1, 1} = false
```

Doesn't seem right 😬

Reply:
I have update the function logic and added mentioned test cases, thanks!